docs/concepts.rst

```diff
@@ -0,0 +1,68 @@
+========
+Concepts
+========
+
+**Kubernetes** is a container orchestrator.
+Over time, Kubernetes has become a standard de facto for the enterprise
+infrastructure management, especially if it based on the microservices.
+
+It provides some basic primitives to orchestrate the application deployments
+on the low level ---such as the pods, jobs, deployments, services, ingresses,
+persistent volumes and volume claims, secrets---
+and allows the Kubernetes cluster to be extended with the arbitrary
+custom resources and custom controllers.
+
+On the top level, it consists of Kubernetes API, through which the users
+talk to Kubernetes, internal storage of the state of the objects (etcd),
+and a collection of the controllers. The command-line tooling (``kubectl``)
+can also be considered as a part of the solution.
+
+----
+
+**Kubernetes controller** is the logic (i.e. the behaviour) behind most
+of the objects, both built-in and added as the extension of Kubernetes.
+For example, the ReplicaSet and Pod creation when a Deployment object
+is created, the rolling version upgrades, and so on.
+
+The main purpose of any controller is to bring the actual state
+of the cluster to the desired state, as expressed with the resources/object
+specifications.
+
+----
+
+**Kubernetes operator** is one kind of the controllers, which orchestrates
+the objects of the specific kind, with some domain logic implemented in it.
+
+The essential difference between the operators and the controllers
+is that the operators are the domain-specific controllers,
+but not all controllers are necessary the operators:
+for example, the built-in controllers for pods, deployments, services, etc;
```

Comment:
it's unclear whether the built-in controllers are operators here :) (the sentence lacks a verb).